### ISSUE TITLE
Linux, obok: undefined variable causes crash

### ISSUE THREAD

**Issue by dcbaker** [apprenticeharper/DeDRM_tools#156]
```txt
Traceback (most recent call last):
  File "calibre_plugins.obok_dedrm.action", line 96, in launchObok
  File "calibre_plugins.obok_dedrm.obok.obok", line 351, in __init__
NameError: global name 'linux_path' is not defined
```

the linux_path variable isn't defined anywhere as far as I can see, even searching for it. I assume that it needs to point to the wine prefix to find the kobo database?

**Comment by apprenticeharper** [comment]
That looks like a good guess. I'm not sure that obok has been properly tested under Linux.

**Comment by norbusan** [comment]
I am running obok on linux quite regularly, without problems, but this is version maybe 3.1.8:
`````
PLUGIN_NAME = 'Obok DeDRM'
PLUGIN_SAFE_NAME = PLUGIN_NAME.strip().lower().replace(' ', '_')
PLUGIN_DESCRIPTION = _('Removes DRM from Kobo kepubs and adds them to the library.')
PLUGIN_VERSION_TUPLE = (3, 1, 8)
`````
And there is nowhere a `linux_path` to be found. Seems that later changes have broken the plugin.

**Comment by norbusan** [comment]
Ok, this and issue 162 are the same, and have been introduced in git commit 3e9516897287a:
`````
@@ -338,6 +341,9 @@ class KoboLibrary(object):
                 self.kobodir = os.path.join(self.kobodir, u"Kobo", u"Kobo Desktop Edition")
             elif sys.platform.startswith('darwin'):
                 self.kobodir = os.path.join(os.environ['HOME'], u"Library", u"Application Support", u"Kobo", u"Kobo Desktop Edition")
+            elif linux_path != None:
+                # Probably Linux, let's get the wine prefix and path to Kobo.
+                self.kobodir = os.path.join(linux_path, u"Local Settings", u"Application Data", u"Kobo", u"Kobo Desktop Edition")
             # desktop versions use Kobo.sqlite
             kobodb = os.path.join(self.kobodir, u"Kobo.sqlite")
             # check for existence of file
`````

Not sure what this change should do, but removeing the elif linux_path clause should fix the problem.

**Comment by apprenticeharper** [comment]
It seems to be an attempt to locate the Kobo Desktop database directory when running under Linux/Wine. I clearly introduced this when merging in fixes from a couple of versions. I'm trying to find where I got this bit from and see how best to fix it. I'm guessing that linux_path is meant to be set to the Wine prefix.

**Comment by apprenticeharper** [comment]
I've checked in a version with those two lines commented out. If anyone has code that actually allows the plugin to work with Kobo Desktop under Wine, I'd be very pleased to include it.

**Comment by das-delta** [comment]
I had the same issue as dcbaker. 

Maybe it helps; here's how I got the plugin to work today:
I'm using Calibre 2.76 under Linux, the newest version of the obok plugin from https://github.com/apprenticeharper/DeDRM_tools/blob/master/Obok_calibre_plugin/obok_plugin.zip, and Kobo Desktop Edition 4.2.6122 (installed with winetricks) in wine.

I modified obok/obok.py: the linux_path error happens in KoboLibrary's `__init__`. 
Important detail: with my setup, the Kobo sqlite database does not reside in the wine directory, but in my home directory. So I ignored the linux_path and simply hardcoded the self.kobodir for my particular setup
```    
elif sys.platform.startswith('linux'):
   self.kobodir = os.environ['HOME']
```
It would probably be good to have that path in a config file later on.

 
Then I restarted calibre 
`calibre-debug -s; calibre-customize -b /home/delta/Downloads/z/; calibre-debug -g`
and was able to select books to import. 

I then got another error (no userkeys found) from the `__getmacaddrs` method because my wine installation has no ipconfig executable installed. 

The if-clause for Darwin can be used for Linux as well:
```
   elif sys.platform.startswith('darwin') or sys.platform.startswith("linux"):
```
 
Hope that helps!
